### PR TITLE
[#3067] feat(doc): Add PR review and merge policy doc

### PR DESCRIPTION
**Pull request by jerryshao**
### What changes were proposed in this pull request?

Add PR review and merge policy into the doc to make it more clear.

### Why are the changes needed?

Fix: #3067 

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

No

### CODE REVIEW

File: MAINTAINERS.md [hunk 1]
```diff
@@ -19,3 +19,65 @@ This document lists the maintainers of the Project. Maintainers may be added onc
...
+like this.
+
+Please use the "Squash and merge" option when merging a PR. This will keep the commit history
+clean and meaningful.
+
+Gravitino use "issue" to track the project progress. So make sure each PR has a related issue, if
+not, please ask the PR author to create one. Unless this PR is a trivial one, like fixing a typo or
+something like that, all PRs should have related issues.
+
+1. Before merging, make sure the PR is approved by **at least** one maintainer.
```
Comment (qqqttt123) [inline MAINTAINERS.md:50]: Maybe we should check the milestone of pull request and issue before we merge the pull request.

File: MAINTAINERS.md [hunk 2]
```diff
@@ -19,3 +19,65 @@ This document lists the maintainers of the Project. Maintainers may be added onc
...
+   PR, like "branch-0.5", "branch-0.6", etc. GitHub Actions will automatically create a backport
+   PR for you.
+6. After PR is merged, please check the related issue:
+   - If the issue is not closed, please close it as fixed manually.
+   - Assign the issue "Assignees" to the PR author.
+   - Starting from 0.6.0, we will use the "labels" to manage the release versions, so please add
+     the corresponding labels to the issue. For example, if the issue is fixed in 0.6.0, please
+     add the label "0.6.0". If the issue is fixed both in 0.6.0 and 0.5.1, please add both labels.
+
+## Policy on backporting bug fixes
```
[the remark this answers is not in the document]
Comment (jerryshao) [reply MAINTAINERS.md:65]: Normally, we don't backport improvement, unless have a consensus that we need to backport it.

File: MAINTAINERS.md (hunk 1 above)
Comment (jerryshao) [reply MAINTAINERS.md:50]: No, PR doesn't have a milestone. In 0.5.0 we use the milestone to manage issues, but due to the limitation of Github that one issue can only have one milestone, so from 0.6.0, I plan to use labels to manage fixed versions, which can support multiple versions.

File: MAINTAINERS.md [hunk 3]
```diff
@@ -19,3 +19,65 @@ This document lists the maintainers of the Project. Maintainers may be added onc
...
+Gravitino maintains several branches for different versions, so backporting bug fixes is a
+common and necessary operation for maintenance release. The decision point is when you have a
+bug fix, and it's not clear whether it is worth backporting or not. Here is the general policy:
+
+For those bug fixes we should backport:
+
+1. Both the bug and fix are well understood and isolated, that is, the fix is not a large change
+   that could introduce new bugs. Code being modified is well tested.
+2. The bug being addressed is high priority or critical to the community.
+3. The backported fix does not vary widely from the main branch fix.
```
Comment (yuqi1129) [inline MAINTAINERS.md:76]: What do you mean by `not vary widely`? Do you mean that the fixed code should align with or be compatible with the main branch?

Comment (jerryshao) [reply MAINTAINERS.md:76]: Yes, if the fixed patch for main branch and backport branch is so different, which means this part of code is changed a lot, the fix will introduce a new bug potentially, we should not encourage to backport.

File: MAINTAINERS.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (qqqttt123) [reply MAINTAINERS.md:65]: OK.

File: MAINTAINERS.md (hunk 1 above)
Comment (qqqttt123) [reply MAINTAINERS.md:50]: OK.

File: MAINTAINERS.md [hunk 4]
```diff
@@ -19,3 +19,65 @@ This document lists the maintainers of the Project. Maintainers may be added onc
 | Clearvive     | Clearvive           | Datastrato       |
 | Cheyne        | ch3yne              | Datastrato       |
 | Jerry Shao    | jerryshao           | Datastrato       |
+
+## Review process
+
+All contributions to the project must be reviewed by **AT LEAST** one maintainer before merging.
+For larger changes, it is recommended to have more than one reviewer. In particular, if you are
+working on a area you are not familiar with, it is recommended to ask for review from that area by
```
Comment (justinmclean) [inline MAINTAINERS.md:27]: an area

File: MAINTAINERS.md [hunk 5]
```diff
@@ -19,3 +19,65 @@ This document lists the maintainers of the Project. Maintainers may be added onc
...
+PRs shall not be merged during active, ongoing discussions, unless they address issues such as
+critical security fixes or public vulnerabilities. Time should be given prior to merging for
+those involved with conversations to be resolved.
+
+## How to merge a pull request
+
+Changes pushed to the main branch on Gravitino cannot be **removed**, that is, we can't do
+force-push to it. So please don't add any test commits or anything like that. If the PR is
+merged by mistake, please using `git revert` to revert the commit, not `git reset` or anything
+like this.
```
Comment (justinmclean) [inline MAINTAINERS.md:41]: -> "like that."

File: MAINTAINERS.md [hunk 6]
```diff
@@ -19,3 +19,65 @@ This document lists the maintainers of the Project. Maintainers may be added onc
...
+
+Please use the "Squash and merge" option when merging a PR. This will keep the commit history
+clean and meaningful.
+
+Gravitino use "issue" to track the project progress. So make sure each PR has a related issue, if
```
Comment (justinmclean) [inline MAINTAINERS.md:46]: "Gravitino use" - > "Gravitino uses"

File: MAINTAINERS.md [hunk 7]
```diff
@@ -19,3 +19,65 @@ This document lists the maintainers of the Project. Maintainers may be added onc
...
+   - If the issue is not closed, please close it as fixed manually.
+   - Assign the issue "Assignees" to the PR author.
+   - Starting from 0.6.0, we will use the "labels" to manage the release versions, so please add
+     the corresponding labels to the issue. For example, if the issue is fixed in 0.6.0, please
+     add the label "0.6.0". If the issue is fixed both in 0.6.0 and 0.5.1, please add both labels.
+
+## Policy on backporting bug fixes
+
+Gravitino maintains several branches for different versions, so backporting bug fixes is a
+common and necessary operation for maintenance release. The decision point is when you have a
```
Comment (justinmclean) [inline MAINTAINERS.md:68]: "maintenance release." -> "maintenance releases."

**Comment by jerryshao** [comment]
Sorry I haven't noticed this issue and merged ahead of time. I will create an separate PR to improve this.

**Pull request merged**